### PR TITLE
Modernize examples

### PR DESCRIPTION
**Pull request by defunctzombie**
### Changelog
None

### Docs
None

### Description
Update examples to use React 18, Eslint v9, and latest versions of @foxglove packages

I deleted the simple-3d-extension. I did not feel that was adding good value to continue having it around. There is nothing special it did with config, setup, etc compared to a typical example panel.

Will also resolve https://github.com/foxglove/create-foxglove-extension/pull/162

### CODE REVIEW

File: examples/call-service-panel-example/eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// @ts-check
+
+const foxglove = require("@foxglove/eslint-plugin");
+const globals = require("globals");
+const tseslint = require("typescript-eslint");
```
Comment (jtbandes) [inline examples/call-service-panel-example/eslint.config.js:5]: typescript-eslint is missing from package.json (I think this applies to all the examples)

File: examples/call-service-panel-example/eslint.config.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+module.exports = tseslint.config({
+  files: ["src/**/*.ts", "src/**/*.tsx"],
```
Comment (jtbandes) [inline examples/call-service-panel-example/eslint.config.js:8]: I think putting this `files` restriction at the top level may mean that this `eslint.config.js` file itself is not linted (and any other `.js` files). Would be interesting to confirm that. Maybe not the end of the world but it feels nice to lint all JS & TS files.

**Review by jtbandes — APPROVED** [review]

File: examples/call-service-panel-example/eslint.config.js (hunk 1 above)
Comment (defunctzombie) [reply examples/call-service-panel-example/eslint.config.js:5]: 🤔 I did run `npm run lint` in each example (installing packages only in that example) and it ran. Maybe something else had it as a transient.

Comment (defunctzombie) [reply examples/call-service-panel-example/eslint.config.js:5]: https://www.npmjs.com/package/@foxglove/eslint-plugin?activeTab=dependencies It is already a dependency of `@foxglove/eslint-plugin`

Comment (jtbandes) [reply examples/call-service-panel-example/eslint.config.js:5]: Sure but don’t we always ensure that we declare dependencies on things we explicitly import? That is what our dependency lint step does in the app repo.

Comment (defunctzombie) [reply examples/call-service-panel-example/eslint.config.js:5]: 🤔 Yea I could see that. I guess my dream is the users worry about fewer dependencies.

**Pull request merged**